src/api/app/controllers/person_controller.rb

```diff
@@ -41,12 +41,12 @@ def get_userinfo
   end
 
   def post_userinfo
-    authorize @user, :update?
-
     login = params[:login]
     # just for permission checking
     User.find_by_login!(login)
 
+    authorize @user, :update?
```

Comment:
Should `set_user` catch this?

Reply:
I don't understand this change either @danidoni. Do you have a scenario where it is behaving wrongly, failing or throwing some exception? To be in the safe side, you could use `find_by!` on `set_user` but TBH, the original code LGTM.

Reply:
Yes, in the code before this change you could send a `login` parameter for a non-existent user (say `"captainreynolds"`). The `set_user` will fill `@user` with `nil`, and then the `authorize @user, :update?` will fail with 
```Pundit::NotDefinedError (unable to find policy `NilClassPolicy` for `nil`)```.
I will try to come up with a failing test.

Reply:
> Should `set_user` catch this?

No, the `set_user` [is not "explosive"](https://github.com/openSUSE/open-build-service/blob/a07fc3ff491c00984049f2261b59a5ffd52f69b6/src/api/app/controllers/person_controller.rb#L265), it will just return `nil`.

Reply:
Well *why* isn't it "explosive"?

Reply:
I've changed it to throw an error, and returning the policy code where it was before.